pkg/cloud/client.go

```diff
@@ -23,27 +23,26 @@ import (
 	infrav1 "github.com/aws/cluster-api-provider-cloudstack/api/v1beta1"
 	"github.com/pkg/errors"
 	"gopkg.in/ini.v1"
-	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
 //go:generate mockgen -destination=../mocks/mock_client.go -package=mocks github.com/aws/cluster-api-provider-cloudstack/pkg/cloud Client
 
 type Client interface {
-	GetOrCreateCluster(*infrav1.CloudStackCluster) error
-	GetOrCreateVMInstance(*infrav1.CloudStackMachine, *capiv1.Machine, *infrav1.CloudStackCluster, string) error
-	ResolveVMInstanceDetails(*infrav1.CloudStackMachine) error
-	DestroyVMInstance(*infrav1.CloudStackMachine) error
-	AssignVMToLoadBalancerRule(*infrav1.CloudStackCluster, string) error
+	ClusterIface
+	VMIface
```

Comment:
Slowly moving to use individual interfaces.

Reply:
whoa, this looks interesting. Can you explain what wizardry is happening here?

Reply:
Just interface composition. : )